### ISSUE TITLE
Avoid hard-coded URI

### ISSUE THREAD

**Issue by opticron** [socketio/socket.io-client-cpp#198]
The hard-coding of the Socket.io URI silently breaks users of this library that pass in a full URI. 

https://github.com/socketio/socket.io-client-cpp/blob/6063cb1d612f6ca0232d4134a018053fb8faea20/src/internal/sio_client_impl.cpp#L218

I'll see if I can whip up a patch for this when I get some time.

**Comment by divmgl** [comment]
Yeah, this is a pain. I believe there are multiple threads related to this open currently. I'll also look into making a pull request.
